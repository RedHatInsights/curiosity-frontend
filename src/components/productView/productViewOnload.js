import React from 'react';
import { Card, CardBody } from '@patternfly/react-core';
import { BinocularsIcon } from '@patternfly/react-icons';
import { useProductOnload } from './productViewContext';
import { ErrorMessage } from '../errorMessage/errorMessage';
import { MessageView } from '../messageView/messageView';
import { translate } from '../i18n/i18n';

/**
 * @memberof ProductView
 * @module ProductViewOnload
 */

/**
 * Product conditional configuration loading.
 *
 * @param {object} props
 * @param {React.ReactNode} props.children
 * @param {translate} [props.t=translate]
 * @param {useProductOnload} [props.useProductOnload=useProductOnload]
 * @returns {JSX.Element}
 */
const ProductViewOnload = ({ children, t = translate, useProductOnload: useAliasProductOnload = useProductOnload }) => {
  const { isReady, error, message, productId } = useAliasProductOnload();

  return (
    (error && (
      <Card isPlain className="curiosity-card curiosity-error__product-onload">
        <CardBody className="curiosity-card__body">
          <ErrorMessage
            message={message}
            title={t('curiosity-view.error', { product: productId, context: ['title', 'onload'] })}
          />
        </CardBody>
      </Card>
    )) ||
    (isReady && children) || (
      <MessageView
        pageTitle="&nbsp;"
        message={t('curiosity-view.pending', { context: 'description' })}
        icon={<BinocularsIcon />}
      />
    )
  );
};

export { ProductViewOnload as default, ProductViewOnload };
