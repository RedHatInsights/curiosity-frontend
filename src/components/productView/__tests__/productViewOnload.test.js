import React from 'react';
import { ProductViewOnload } from '../productViewOnload';

describe('ProductViewOnload Component', () => {
  it('should render a basic component', async () => {
    const props = {
      useProductOnload: () => ({ isReady: true })
    };
    const component = await shallowComponent(<ProductViewOnload {...props}>lorem ipsum</ProductViewOnload>);
    expect(component).toMatchSnapshot('basic');
  });

  it('should handle display error responses', async () => {
    const props = {
      useProductOnload: () => ({ error: true, message: 'error message', productId: 'mockProductId' })
    };
    const component = await shallowComponent(<ProductViewOnload {...props}>lorem ipsum</ProductViewOnload>);
    expect(component).toMatchSnapshot('error');
  });

  it('should handle pending responses', async () => {
    const props = {
      useProductOnload: () => ({ isReady: false })
    };
    const component = await shallowComponent(<ProductViewOnload {...props}>lorem ipsum</ProductViewOnload>);
    expect(component).toMatchSnapshot('pending');
  });
});
