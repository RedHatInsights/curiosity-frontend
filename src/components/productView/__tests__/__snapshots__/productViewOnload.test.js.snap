// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductViewOnload Component should handle display error responses: error 1`] = `
<Card
  className="curiosity-card curiosity-error__product-onload"
  isPlain={true}
>
  <CardBody
    className="curiosity-card__body"
  >
    <ErrorMessage
      message="error message"
      title="t(curiosity-view.error_title, {"product":"mockProductId","context":"onload"})"
    />
  </CardBody>
</Card>
`;

exports[`ProductViewOnload Component should handle pending responses: pending 1`] = `
<MessageView
  icon={<BinocularsIcon />}
  message="t(curiosity-view.pending, {"context":"description"})"
  pageTitle="Â "
/>
`;

exports[`ProductViewOnload Component should render a basic component: basic 1`] = `"lorem ipsum"`;
